name: taxonomy-manual

on:
  workflow_dispatch:
    inputs:
      project:
        description: 'Project to test'
        required: true
      start_from:
        description: 'start index'
        required: true
      end_to:
        description: 'end index'
        required: true
jobs:
  taxonomy:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: docker-practice/actions-setup-docker@master
      - name: Set up Python 3.8
        uses: actions/setup-python@v2
        with:
          python-version: '3.8'
      - name: Test taxonomy
        run: |
          python -m pip install --upgrade pip
          pip install -r requirements_dev.txt
          mkdir mydir
      - name: Run manual tests
        run: PYTHONPATH=bugscpp/ python -m pytest --basetemp=mydir --auto-cleanup --uid=1001 --start-from=${{ github.event.inputs.start_from }} --end-to=${{ github.event.inputs.end_to }} --project=${{ github.event.inputs.project }} --no-skip tests/taxonomy/test_taxonomy.py
        env:
          # Set BUGSCPP_TEST_TAXONOMY=1
          # In docker testing, test broken when pull image.
          BUGSCPP_TEST_TAXONOMY: 1

